integration/testnet/util.go

```diff
@@ -105,3 +108,50 @@ func WriteJSON(path string, data interface{}) error {
 	err = ioutil.WriteFile(path, marshaled, 0644)
 	return err
 }
+
+// rootProtocolJsonWithoutAddresses strips out all node addresses from the root protocol json file specified as srcFile
+// and creates the dstFile with the modified contents
+func rootProtocolJsonWithoutAddresses(srcfile string, dstFile string) error {
+
+	data, err := io.ReadFile(filepath.Join(srcfile))
+	if err != nil {
+		return err
+	}
+
+	var rootSnapshot inmem.EncodableSnapshot
+	err = json.Unmarshal(data, &rootSnapshot)
+	if err != nil {
+		return err
+	}
+
+	removeAddress := func(ids flow.IdentityList) {
+
+		for _, identity := range ids {
+			identity.Address = ""
+			//TODO: following doesn't work and Address is still retained
```

Comment:
Tried a few different things 👇 to influence the `json.Marshal` call to skip fields dynamically but none of them have  worked so far.
1.  This 3rd party library to manager tags - https://github.com/sevlyar/retag
2. Couple of other suggestions from here - https://stackoverflow.com/questions/17306358/removing-fields-from-struct-or-hiding-them-in-json-response.

Reply:
Here is one idea that I think could help:
https://play.golang.org/p/5gjkRpGngkK

Reply:
You could also add some logic to the existing `MarshalJSON` [method](https://github.com/onflow/flow-go/blob/1bb8a60dd79de3fa0e7ac327b466f12f3c1430fd/model/flow/identity.go#L115-L125) to skip the `Address` field if it is empty:
```go
if id.Address == "" {
  return json.Marshal(identityWithoutAddress) // will need a new type here, like in Francois' example
} else {
  return json.Marshal(identity) // do the same thing it's currently doing
}
```